File: examples/optimizations/primes.py
from typing import List

def finder(amount: int) -> List[int]:
    primes = []
    found = 0
    number = 2

    while found < amount:
        for x in primes[:found]:
            if number % x == 0:
                break
        else:
            primes.append(number)
            found += 1
        number += 1
    
    return primes

File: examples/optimizations/primes.c
#include <stdio.h>
#include <stdlib.h>

void finder(int amount, int *primes) {
    int found = 0;
    int number = 2;
    int is_prime, i;

    while (found < amount) {
        is_prime = 1;
        for (i = 0; i < found; i++) {
            if (number % primes[i] == 0) {
                is_prime = 0;
                break;
            }
        }
        if (is_prime) {
            primes[found++] = number;
        }
        number++;
    }
}

// gcc -shared -fpic -o libprimes.so primes.c

